### ISSUE TITLE
Support additional metrics

### ISSUE THREAD

**Issue by osma** [NatLibFi/Annif#141]
Sklearn has a few more metrics for multilabel evaluation. Especially interesting are metrics that can work with large result sets without needing to specify a limit.

To use these we probably need to turn the results into vectors instead of the current approach which evaluates one document at a time.

The evaldir and optimize commands should also show how many documents were evaluated.

**Comment by osma** [comment]
Here are the [multilabel metrics](http://scikit-learn.org/stable/modules/model_evaluation.html#multilabel-ranking-metrics) available in sklearn. Thoughts:

* Coverage error: probably not useful because most backends don't return the full set of labels
* Label ranking average precision: looks very useful
* Ranking loss: maybe not as useful as LRAP. Scale is inverse (0 is best). But should be simple to implement together with LRAP

**Comment by osma** [comment]
Hmm. Also LRAP and ranking loss may require full results...need to test.

**Comment by osma** [comment]
Tested all three measures and they all worked, even when given incomplete sets of labels. So all could be used. The problem is just that the current way of calculating statistics (one document at a time) is very different from what sklearn expects (vectors, vectors...), so it needs to be re-architected to be able to make use of sklearn measures (and then we could maybe also use sklearn for precision, recall, F-measure etc)

**Comment by osma** [comment]
What remains to be done:

- [ ] add micro- and macroaveraged precision/recall/f1-score
- [ ] add LRAP score
- [ ] add ranking loss score
- [ ] evaldir and optimize should show how many documents were evaluated

**Comment by osma** [comment]
Good enough now.
